### PR TITLE
Navbar refactor

### LINKED ISSUE #60: Refactor NavBar in Application View
We can refactor the navbar in the application view using a partial for each type of user.

If you start this ping me before! I have some ideas I wanna share before we dive into this one.

### PR DESCRIPTION
**Pull request by lkriffell**
Created four partials: One for each type of user
closes #60

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -11,24 +11,15 @@
...
+        <%= render partial: './users/visitor_nav' %>
+      <% elsif current_user.role == "regular" %>
+        <%= render partial: './users/user_nav' %>
       <% elsif current_user.admin? %>
-        <%= link_to "All Users", "/admin/users" %>
-        <%= link_to "Dashboard", "/admin/dashboard" %>
+        <%= render partial: './users/admin_nav' %>
       <% elsif current_user.merchant? %>
-        <%= link_to "Dashboard", "/merchant/dashboard" %>
+        <%= render partial: './users/merchant_nav' %>
```
Comment (AngelaGuardia) [inline app/views/layouts/application.html.erb:21]: If you want to, we could clean this up even more with guard statements - we'd end up with only four lines in our navbar!

**Review by AngelaGuardia — APPROVED** [review]
Love how this refactor turned out. Thanks for doing this Logan! 

@KieraAllen This is a good candidate to show how our code progressed and got better over time in our README

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (lkriffell) [reply app/views/layouts/application.html.erb:21]: Like put these statements ```<% elsif current_user.admin? %>``` into the partial itself?

File: app/views/users/_admin_nav.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<% if current_user && current_user.admin? %>
```
Comment (lkriffell) [inline app/views/users/_admin_nav.html.erb:1]: Had to add a check that current_user is not nil before checking if current_user is an admin, merchant, or user. Seems like there's even more room for refactoring this but I'm not sure what that would be

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (AngelaGuardia) [reply app/views/layouts/application.html.erb]: No they would remain in this view and be like this:
```
<%= render partial: './users/user_nav' if current_user.regular? %>
<%= render partial: './users/admin_nav' if current_user.admin? %>
```
... and so on

File: app/views/users/_admin_nav.html.erb (hunk 2 above)
Comment (AngelaGuardia) [reply app/views/users/_admin_nav.html.erb:1]: Maybe your `current_admin?` `current_merchant?` methods.

**Review by KieraAllen — APPROVED** [review]

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -11,10 +11,13 @@
...
-      <%= render partial: './users/user_nav' %>
-      <%= render partial: './users/admin_nav' %>
-      <%= render partial: './users/merchant_nav' %>
+      <% if current_user.nil? %>
+        <%= render partial: './users/visitor_nav' %>
+      <% else %>
+        <%= render partial: './users/user_nav' if current_user.role == "regular" %>
+        <%= render partial: './users/admin_nav' if current_user.admin? %>
+        <%= render partial: './users/merchant_nav' if current_user.merchant? %>
+      <% end %>
```
Comment (AngelaGuardia) [inline app/views/layouts/application.html.erb]:
```suggestion
        <%= render partial: './users/visitor_nav' if current_user.nil? %>
        <%= render partial: './users/user_nav' if current_user.role == "regular" %>
        <%= render partial: './users/admin_nav' if current_user.admin? %>
        <%= render partial: './users/merchant_nav' if current_user.merchant? %>
```

**Review by AngelaGuardia — APPROVED** [review]

**Pull request merged**